CondorMonitoring/JobCounter.py

```diff
@@ -589,72 +604,99 @@ def main():
+                jobs = schedd.xquery('true', classAds[schedd_type])
             except RuntimeError:
                 exc_type, exc_value, exc_traceback = sys.exc_info()
                 print "ERROR: Failed to query schedd %s with:\n" % schedd_name
                 print repr(traceback.format_exception(exc_type, exc_value, exc_traceback))
+            try:
+                for job in jobs:
 
-            for job in jobs:
-
-                jobId = str(job['ClusterID']) + '.' + str(job['ProcId'])
-                status = int(job['JobStatus'])
-
-                if 'WMAgent_SubTaskName' not in job:
-                    print 'I found a job not coming from WMAgent: %s' % jobId
-                    continue
-
-                workflow = job['WMAgent_SubTaskName'].split('/')[1]
-                task = job['WMAgent_SubTaskName'].split('/')[-1]
-                jType = job['CMS_JobType']
-                try:  # it can be an ExprTree
-                    cpus = int(job['RequestCpus'])
-                except TypeError:
-                    cpus = 1
-                siteToExtract = str(job['DESIRED_Sites']).replace(' ', '').split(",")
-
-                if schedd_name in relvalAgents:  # If RelVal job
-                    jType = 'RelVal'
-                elif task == 'Reco':  # If PromptReco job (Otherwise type is Processing)
-                    jType = 'Reco'
-                elif jType not in jobTypes:  # If job type is not standard
-                    jType = jobType(jobId, schedd_name, task)
-
-                siteRunning = job.get('MATCH_EXP_JOBGLIDEIN_CMSSite', '')
-                if siteName(siteRunning):  # If job is currently running
-                    siteToExtract = [siteRunning]
-
-                # Ignore jobs to the T1s from the Tier-0 pool
-                # Avoid double accounting with the global pool
-                if collector_name == tier0_pool[0]:
-                    if not 'T2_CH_CERN_T0' in siteToExtract:
-                        continue
+                    jobId = str(job['ClusterID']) + '.' + str(job['ProcId'])
+                    status = int(job['JobStatus'])
 
-                # Ignore jobs to T2_CH_CERN_T0 from the global pool
-                # Avoid double accounting with the Tier-0 pool
-                if collector_name == global_pool[0]:
-                    if 'T2_CH_CERN_T0' in siteToExtract:
+                    if jobKeys[schedd_type]['taskname'] not in job:
+                        # CRAB has a lot of these which are not tagged and they are task processes.
+                        # They should not be accounted for anything.
+                        #print 'I found a job coming from %s, but it does not have needed classads. JobID %s' % (schedd_type, jobId)
+                        continue
+                    workflow = None
+                    task = None
+                    if schedd_type == 'prodschedd':
+                        workflow = job['WMAgent_SubTaskName'].split('/')[1]
```

Comment:
1 vs 2 attributes? Better stick to one, which has and TaskName and SubTaskName. You can update it by yourself if you think differently